embeddings/model/lightning_model.py

```diff
@@ -21,7 +22,13 @@ def __init__(
         self.predict_subset = predict_subset
 
     def execute(self, data: pl.LightningDataModule) -> Dict[str, nptyping.NDArray[Any]]:
-        self.trainer.fit(self.task, data)
+        try:
+            self.trainer.fit(self.task, data)
+        except Exception as e:
+            del self.trainer
```

Comment:
should this be put here or somewhere in hps related code? would teardown method work here as well?

Reply:
It should be placed here as error while training may occur also while not hps related code. As I understand teardown correctly it's called every time after fit, not only while error occurs.